File: rb/features.rb
# Openwhyd SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OpenwhydFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpenwhydBaseFeature.new
    when "ratelimit"
      OpenwhydRatelimitFeature.new
    when "retry"
      OpenwhydRetryFeature.new
    when "test"
      OpenwhydTestFeature.new
    when "timeout"
      OpenwhydTimeoutFeature.new
    else
      OpenwhydBaseFeature.new
    end
  end
end

File: py/features.py
# Openwhyd SDK feature factory

from feature.base_feature import OpenwhydBaseFeature
from feature.test_feature import OpenwhydTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenwhydBaseFeature(),
        "test": lambda: OpenwhydTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
